client/X11/xf_gdi.c

```diff
@@ -1122,14 +1122,14 @@ void xf_gdi_surface_bits(rdpContext* context, SURFACE_BITS_COMMAND* cmd)
 	RFX_MESSAGE* message;
 	xfContext* xfc = (xfContext*) context;
 
-	xf_lock_x11(xfc, FALSE);
```

Comment:
Here I move the xf_lock_x11 after bitmap data decode so that other thread which need X display could run on other CPU while time-consuming bitmap decode task is running.